cmd/segment-reaper/detect.go

```diff
@@ -25,9 +34,126 @@ var (
 		DatabaseURL string `help:"the database connection string to use" default:"postgres://"`
 		From        string `help:"begin of date range for detecting zombie segments" default:""`
 		To          string `help:"end of date range for detecting zombie segments" default:""`
+		File        string `help:"location of file with report" default:"detect_result.out"`
 	}
 )
 
+// Cluster key for objects map
+type Cluster struct {
+	projectID string
+	bucket    string
+}
+
+// Object represents object with segments
+type Object struct {
+	// TODO verify if we have more than 64 segments for object in network
+	segments uint64
+
+	expectedNumberOfSegments byte
+
+	hasLastSegment bool
+	// if skip is true segments from object should be removed from memory when last segment is found
+	// or iteration is finished, mark it as true if one of the segments from this object is newer
+	// then specified threshold
+	skip bool
+}
+
+// ObjectsMap map that keeps objects representation
+type ObjectsMap map[Cluster]map[string]*Object
```

Comment:
What is the `string` key in the map? If it is the object path, can we it be of type `storj.Path` to be more obvious?

Reply:
yes, its everything after bucket name in path. I will change it to `storj.Path`